mantidproject/mantid
# Handle empty alarm log files

**Description of work.**
Added check for empty ICPalarm log files when loading raw data files to prevent LoadLog throwing an exception when it attempts to process an empty log file. The user will no longer see a worrying warning message for a situation that is normal.

**To test:**
Do not test on Windows (see #31986 for reason).
- Change the log level to `information` by right clicking the messages widget
- Run Load on /archive/NDXINES/Instrument/data/cycle_20_3/INS17770.raw
- Verify that there are no warning messages
- Verify that the following information message is present:
`Skipping empty log file: /path/to/file/INS17770_ICPalarm.txt`

Fixes #31605

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
